### PR TITLE
[MINOR][DOCS] Fix typo in documents

### PR DESCRIPTION
**Pull request by kiszk**
### What changes were proposed in this pull request?
Fixed typo in `docs` directory and in `project/MimaExcludes.scala`

### Why are the changes needed?
Better readability of documents

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
No test needed

### CODE REVIEW

File: project/MimaExcludes.scala [hunk 1]
```diff
@@ -1422,7 +1422,7 @@ object MimaExcludes {
       ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.SparkEnv.getThreadLocal"),
       ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.mllib.rdd.RDDFunctions.treeReduce"),
       ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.mllib.rdd.RDDFunctions.treeAggregate"),
-      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.mllib.tree.configuration.Strategy.defaultStategy"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.mllib.tree.configuration.Strategy.defaultStrategy"),
```
Comment (HyukjinKwon) [inline project/MimaExcludes.scala:1425]: I think `defaultStategy` is correct. It was deprecated at SPARK-9609 and removed at SPARK-14089. `defaultStrategy` seems fine.

File: docs/ml-migration-guide.md [hunk 2]
```diff
@@ -281,7 +281,7 @@ Several deprecated methods were removed in the `spark.mllib` and `spark.ml` pack
 * `weights` in `LinearRegression` and `LogisticRegression` in `spark.ml`
 * `setMaxNumIterations` in `mllib.optimization.LBFGS` (marked as `DeveloperApi`)
 * `treeReduce` and `treeAggregate` in `mllib.rdd.RDDFunctions` (these functions are available on `RDD`s directly, and were marked as `DeveloperApi`)
-* `defaultStategy` in `mllib.tree.configuration.Strategy`
+* `defaultStrategy` in `mllib.tree.configuration.Strategy`
```
Comment (HyukjinKwon) [inline docs/ml-migration-guide.md:284]: This one too. It was deprecated at SPARK-9609.

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise.

File: project/MimaExcludes.scala (hunk 1 above)
Comment (kiszk) [reply project/MimaExcludes.scala:1425]: Thank you for letting me know the history. I will revert these two lines.

**Comment by kiszk** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Pull request closed without merging**